core/src/main/scala/kafka/server/DelayedOperation.scala

```diff
@@ -288,58 +282,49 @@ class DelayedOperationPurgatory[T <: DelayedOperation](purgatoryName: String,
   }
 
   /**
-   * A linked list of watched delayed operations based on some key
+   * A vector of watched delayed operations for a specific key
    */
   private class Watchers(val key: Any) {
 
-    private[this] val operations = new LinkedList[T]()
+    private[this] var operations = Vector[T]()
 
-    def watched: Int = operations synchronized operations.size
+    def watched: Int = synchronized(operations.size)
```

Comment:
Similar to @becketqin's comment on readability. It might be more immediately obvious what this is doing with
```scala
def watched: Int = synchronized {
  operations.size
}